android/src/app/organicmaps/car/screens/search/SearchScreen.java

```diff
@@ -35,14 +50,108 @@ public Template onGetTemplate()
     final SearchTemplate.Builder builder = new SearchTemplate.Builder(this);
     builder.setHeaderAction(Action.BACK);
     builder.setShowKeyboardByDefault(false);
-    if (mSearchText.isEmpty() || mResults == null)
-      loadRecents();
-    builder.setItemList(mResults);
-    builder.setInitialSearchText(mSearchText);
+    if (mQuery.isEmpty() && mResults == null)
+    {
+      if (!loadRecents())
+        builder.setShowKeyboardByDefault(true);
+    }
+    if (!mQuery.isEmpty() && mResults != null && !mResults.getItems().isEmpty())
+      builder.setActionStrip(createActionStrip());
+    if (mResults == null)
+      builder.setLoading(true);
+    else
+      builder.setItemList(mResults);
+    builder.setInitialSearchText(mQuery);
+    builder.setSearchHint(getCarContext().getString(R.string.search));
+    return builder.build();
+  }
+
+  @Override
+  public void onSearchTextChanged(@NonNull String searchText)
+  {
+    if (mQuery.equals(searchText))
+      return;
+    mQuery = searchText;
+    mResults = null;
+
+    SearchEngine.INSTANCE.cancel();
+
+    if (mQuery.isEmpty())
+    {
+      invalidate();
+      return;
+    }
+
+    final MapObject location = LocationHelper.INSTANCE.getMyPosition();
+    final boolean hasLocation = location != null;
+    final double lat = hasLocation ? location.getLat() : 0;
+    final double lon = hasLocation ? location.getLon() : 0;
+
+    SearchEngine.INSTANCE.search(getCarContext(), mQuery, false, System.nanoTime(), hasLocation, lat, lon);
```

Comment:
Does it make sense to overload this function and simplify the calling code here and above in a similar call?

Reply:
I'm not following you. What's the problem here? What can be simplified?

Reply:
Create overload without haslocation and lat lon if there is no location.